src/view/components/countries/CountryDropdown.tsx

```diff
@@ -0,0 +1,71 @@
+import { Combobox, Transition } from "@headlessui/react";
+import React from "react";
+import {
+    COUNTRY_CO2_EMISSIONS_GRAMS_PER_GB,
+    CountryName,
+} from "../../../data/constants/CountryEmissions";
+
+type CountryDropdownType = {
+    addSelectedCountry: (country: CountryName) => void;
+    selectedCountries: Map<CountryName, number>;
+};
+
+export const CountryDropdown = ({
+    addSelectedCountry,
+    selectedCountries,
+}: CountryDropdownType) => {
+    const filteredCountries = Object.keys(
+        COUNTRY_CO2_EMISSIONS_GRAMS_PER_GB
+    ).filter((country) => !selectedCountries.has(country as CountryName));
+
+    const [query, setQuery] = React.useState("");
+
+    const displayedCountries =
+        query === ""
+            ? filteredCountries
+            : filteredCountries.filter((country) =>
+                  country
+                      .toLowerCase()
+                      .replace(/\s+/g, "")
```

Comment:
- Why is regex necessary in this case?

Reply:
True, it is unnecessary on this line! I'll remove it in the next commits.

Reply:
On second thought, I think it is necessary so that you can implement this behaviour (correct me if I am wrong): 

filteredCountries: United Kingdom --> unitedkingdom, United States --> unitedstates

query: uN iTe --> unite which should match "United Kingdom" and "United States"